src/emqx_slow_subs/emqx_message_elapsed_stats.erl

```diff
@@ -0,0 +1,93 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_message_elapsed_stats).
+
+%% API
+-export([new/3, update/2, check_expire/3]).
+
+-define(NOW, erlang:system_time(millisecond)).
+-define(MINIMUM_INSERT_INTERVAL, 1000).
+-define(MINIMUM_THRESHOLD, 100).
+
+-type stats() :: #{ clientid := emqx_types:clientid()
```

Comment:
it is great that we created a new module to encapsulate the data structure.
however it's not so great to directly access the internals from other modules

consider doing this:
* define it as an `-opaque() :: `, and export it in `export_type`
* add getter APIs to retrieve field values from outside of this module.

Reply:
clientid field is redundant.
this module does not need to have this information.